src/testing/support/assertRender.ts

```diff
@@ -98,9 +99,67 @@ function formatNode(node: WNode | VNode, tabs: any): string {
 	return `v("${node.tag}", ${properties}`;
 }
 
+const assertionWidgets = [
+	{
+		type: Ignore,
+		value(actual: DNode, expected: DNode) {
+			const node = actual ? actual : expected;
+			return [actual, node];
+		}
+	}
+];
+
+function isNode(node: any): node is VNode | WNode {
+	return isVNode(node) || isWNode(node);
+}
+
+function decorate(actual: DNode | DNode[], expected: DNode | DNode[]): [DNode[], DNode[]] {
+	actual = Array.isArray(actual) ? actual : [actual];
+	expected = Array.isArray(expected) ? expected : [expected];
+	let actualDecoratedNodes = [];
+	let expectedDecoratedNodes = [];
+	const length = actual.length > expected.length ? actual.length : expected.length;
+	for (let i = 0; i < length; i++) {
+		let actualNode = actual[i];
+		let expectedNode = expected[i];
+
+		if (isWNode(expectedNode)) {
+			[actualNode, expectedNode] = assertionWidgets.reduce(
```

Comment:
this part is somewhat confusing to me. I can follow what it is doing, but it isn't clear to me why it i doing what it is doing. If you are changing the `actualNode` and `expectedNode` references, does it really make sense to use `reduce` here? I think a `forEach` might end up being simpler.

Reply:
I agree, i just removed the loop completely in the end as the indirection wasn't worth it.